### ISSUE TITLE
[ Suites ] Suites de nombres complexes

### ISSUE THREAD

**Issue by fmOOmf** [UpsilonNumworks/Upsilon#165]
Bonjour à toute l'équipe

_(En préambule :  le comportement décrit ici est identique pour le firmware Epsilon E17)_
J'ai défini la suite de nombres complexes suivante : Un = (1+i)^n
Puis, j'affiche les premiers éléments à l'aide de l'onglet "Tableau".

Il apparaît que l'application "suite" sait calculer les premiers éléments, mais ne semble pas gérer l'affichage lorsque le résultat est un nombre complexe. Dans ce cas, il est affiché "undef".
Le nombre attendu est réel pour les indices n=0, 4, 8, 12 ... Le tableau affiche bien 1 (pour n=0),  -4 (pour n=4), mais rien pour n=8 (16 attendu). 

Amélioration proposée : 
Afficher les résultats en nombre complexes. 
Si ce n'est pas possible, indiquer "nombre complexe" plutôt que "undef"  pour que l'élève comprenne que le résultat peut être calculé, mais ne peut être affiché.
Vérifier pourquoi rien n'est affiché pour n = 8

Copies d'écran : 
La suite : 
![image](https://user-images.githubusercontent.com/98671961/154937760-d2d9ae5f-1c2c-4617-b56d-b2d954a14d8c.png)

Les résultats obtenus : 
![image](https://user-images.githubusercontent.com/98671961/154942102-2c72272f-fe72-47ae-8b6e-5e536b355ed4.png)

**Comment by Lauryy06** [comment]
Effectivement, je pense que la calculatrice devrait afficher unreal au lieu de undef, et être capable d'afficher quand même les valeurs réelles de (u).

**Comment by fmOOmf** [comment]
Bonjour

Ici aussi, le label "Bug" est sévère si les suites n'ont pas été prévues à la base pour traiter les nombres complexes.
Il vaut mieux effectivement afficher "unreal" (ou "undef" à la limite) plutôt que les parties seules réelles,  (ce qui induirait les élèves en erreur).
Afficher les nombres complexes peut vite prendre de la place, mais serait une évolution très intéressante.

**Comment by Lauryy06** [comment]
A nouveau, étant donné que le résultat attendu est au moins "unreal" au lieu de "undef", et que certaines valeurs réelles ne sont pas affichées, je pense que l'on peut considérer ceci comme un bug.
